CMakeLists.txt

```diff
@@ -288,7 +288,7 @@ if(NOT SNMALLOC_HEADER_ONLY_LIBRARY)
 
   if(NOT (DEFINED SNMALLOC_LINKER_FLAVOUR) OR ("${SNMALLOC_LINKER_FLAVOUR}" MATCHES "^$"))
     # Linker not specified externally; probe to see if we can make lld work
-    set(CMAKE_REQUIRED_LINK_OPTIONS -fuse-ld=lld)
+    set(CMAKE_REQUIRED_LINK_OPTIONS -fuse-ld=lld -Wl,--icf=all)
```

Comment:
Why did you remove this?  It makes the library smaller as there are a lot of names for the same function in the API surface (e.g. lots of new overloads and malloc can all be folded into a single thing).

Reply:
FYI: this change increases the size of `libsnmallocshim.so` by about 10%.

Reply:
I didn't remove it, I added it to the check that enables it.  Without this, we now have CI that triggers the reason that I have to carry a local patch for snmalloc to disable this: If you compile with GCC, the `-fuse-ld=lld` test fails (GCC consumes but silently ignores the flag?  Or just warns and doesn't error?), but then GCC passes this flag to ld.bfd and it fails to link.

Reply:
Thanks. I miss read the diff.